### PR TITLE
Added correct isinf handling for Integral tensors

### PR DESCRIPTION
**Pull request by frankz-ai**
Currently torch.isinf on integral tensor will raise RuntimeError: value cannot be converted to type int16_t without overflow: inf. This pr will suppress the error and return false(0) for all integral tensors. The behavior will also be consistent with np.isinf

### CODE REVIEW

File: test/test_torch.py [hunk 1]
```diff
@@ -5317,8 +5317,18 @@ def test_isfinite_int(self):
         self.assertEqual(torch.isfinite(x), torch.ByteTensor([1, 1, 1]))
 
     def test_isinf(self):
-        x = torch.Tensor([1, inf, 2, -inf, nan])
-        self.assertEqual(torch.isinf(x), torch.ByteTensor([0, 1, 0, 1, 0]))
+        t1 = torch.Tensor([1, inf, 2, -inf, nan])
```
Comment (zou3519) [inline test/test_torch.py]: Could you add a test to test/test_cuda.py as well? It should reference the test torch mixin: https://github.com/pytorch/pytorch/blob/3da4a0473368e9ac18376a9797c46c28cf0bb45d/test/test_cuda.py#L938

**Review by zou3519 — APPROVED** [review]
LGTM! Could use a CUDA test but it's not entirely necessary because the code just goes through torch.zeros_like

**Comment by zou3519** [comment]
@flashhack please fix the python lint (see the travis test output)

**Comment by soumith** [comment]
fixes https://github.com/pytorch/pytorch/issues/15123

**Pull request closed without merging**